### PR TITLE
Fix top k parameter

### PR DESCRIPTION
**Pull request by andyfengHKU**
# Description

Fix a segfault when trying to bind parameter in Top K limit.

### CODE REVIEW

File: test/main/connection_test.cpp [hunk 1]
```diff
@@ -199,26 +199,49 @@ TEST_F(ApiTest, PrepareWithSkipAndLimit) {
...
+    auto prepared = conn->prepare("MATCH (p:person) RETURN p.ID ORDER BY p.ID limit $lm");
+    auto result = conn->execute(prepared.get(), std::make_pair(std::string{"lm"}, 5));
+    ASSERT_TRUE(result->isSuccess());
+    std::vector<std::string> expectedResult = {"0", "2", "3", "5", "7"};
+    ASSERT_EQ(TestHelper::convertResultToString(*result), expectedResult);
+    result = conn->execute(prepared.get(), std::make_pair(std::string{"lm"}, 2));
+    ASSERT_TRUE(result->isSuccess());
+    expectedResult = {"0", "2"};
+    ASSERT_EQ(TestHelper::convertResultToString(*result), expectedResult);
+}
```
Comment (acquamarin) [inline test/main/connection_test.cpp:226]: Can we add a test case when the user pass in a negative number?

**Review by acquamarin — APPROVED** [review]
